OpenRA.Mods.Common/Traits/SupportPowers/ProduceActorPower.cs

```diff
@@ -58,7 +58,7 @@ public override void Activate(Actor self, Order order, SupportPowerManager manag
 
 			var info = Info as ProduceActorPowerInfo;
 			var sp = self.TraitsImplementing<Production>()
-				.FirstOrDefault(p => p.Info.Produces.Contains(info.Type));
+				.FirstOrDefault(p => !p.IsTraitDisabled && !p.IsTraitPaused && p.Info.Produces.Contains(info.Type));
```

Comment:
A better approach would be to loop over these until the production succeeds instead of picking one and then doing nothibg id it fails (e.g. door is blocked)

Reply:
Here is checking for Production traits of one actor. SW is for some reason always activated from the topmost (and leftmost) actor, if it can produce or not.

There are 2 problem, iirc original game has Primary buildings for palaces too. ProduceActorPower ignore primarity. Palaces on our D2K mod don't have PrimaryBuilding: anyway, but i think should be added.

If first actor fails. It should look for other ones. It doesn't, i'm trying to find the responsible code, but couldn't yet.

This check probably should be here (considering we allow multipile Production: traits). But doesn't fix the whole issue.

Reply:
Take a look at and copy what ClassicProductionQueue.BuildUnit does.